core/src/main/java/io/grpc/transport/AbstractClientStream.java

```diff
@@ -45,9 +47,12 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import javax.annotation.concurrent.NotThreadSafe;
+
 /**
  * The abstract base class for {@link ClientStream} implementations.
  */
+@NotThreadSafe
```

Comment:
@ejona86  Alright.  I am going to revert this change since it doesn't look like this will be resolved.